dependencies.yaml

```
@@ -266,6 +268,16 @@ dependencies:
               arch: aarch64
             packages:
               - cupy-cuda11x -f https://pip.cupy.dev/aarch64 # TODO: Verify that this works.
+  test_cpp:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.8"
+            packages:
+              - cuda-sanitizer-api=11.8.86
```

This is a very specific pinning, any reason we can't do a >=?

---

This was stolen from here: https://github.com/rapidsai/cudf/pull/12530/files#diff-5475a6d76de4c506ee92cf6f941bba30a6a07b5881cfea531d42a6ec035095a6R278
cc @bdice

---

@vyasr We don't want the compute sanitizer from CUDA 12 to be run with our CUDA 11.8 builds. We need to pin this.

---

We could pin like suggested below, but I wouldn't advise it. We have been pinning exact dependencies on CUDA packages in our conda build recipes like shown here: https://github.com/rapidsai/raft/blob/branch-23.04/conda/recipes/libraft/conda_build_config.yaml
```suggestion
              - cuda-sanitizer-api=11.8.*
```

---

I thought about this myself too and figured it was best left as-is.

The downside to this pin is that we'll need to update the version and corresponding `matrix` values in the `dependencies.yaml` file anytime the `rapidsai/ci:latest` image is updated (which is used [here](https://github.com/rapidsai/cudf/pull/12800/files#diff-245392b692a50c38ecab4381b118862db514035c10983f3bd4f4b7f1f4be4692R35)).

Right now the `latest` tag points to an `11.8` image. When we update that image to CUDA 12 for instance, then `cuda-santizer-api` will no longer get installed and the job will fail.

---

That's correct -- but that's already true for every other part of our CI. We need to keep our conda recipes (and conda build configs) **and** dependencies.yaml matrix values up to date for every CUDA update we make. I don't think this is a new requirement, it's just another instance of things we must already do.

---

I guess I was mostly looking to float forward minor versions while pinning the major version i.e. ` ==11.*` but if we're being consistent here that's fine with me.